### PR TITLE
Avoiding unnecesary allocations at rustc_errors

### PR DESCRIPTION
**Pull request by zzau13**
Simplify the code avoiding allocations with easy alternative

### CODE REVIEW

File: compiler/rustc_errors/src/lib.rs [hunk 1]
```diff
@@ -973,16 +973,14 @@ impl HandlerInner {
...
     fn panic_if_treat_err_as_bug(&self) {
         if self.treat_err_as_bug() {
-            let s = match (self.err_count(), self.flags.treat_err_as_bug.unwrap_or(0)) {
-                (0, _) => return,
-                (1, 1) => "aborting due to `-Z treat-err-as-bug=1`".to_string(),
-                (1, _) => return,
-                (count, as_bug) => format!(
+            match (self.err_count(), self.flags.treat_err_as_bug.unwrap_or(0)) {
+                (1, 1) => panic!("aborting due to `-Z treat-err-as-bug=1`"),
+                (0, _) | (1, _) => (),
```
Comment (lzutao) [inline compiler/rustc_errors/src/lib.rs:978]: Prefer style in rustc. Probably should add this in fmt-rfcs
```suggestion
                (0, _) | (1, _) => {}
```

**Review by davidtwco — APPROVED** [review]
LGTM

**Comment by davidtwco** [comment]
Could you squash the last commit that changes code style?

**Comment by botika** [comment]
> Could you squash the last commit that changes code style?



> Could you squash the last commit that changes code style?

Done, 

Thanks for your time!!

**Review by davidtwco — APPROVED** [review]

**Comment by davidtwco** [comment]
Thanks for the contribution!

@bors r+

**Pull request merged**